src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -695,6 +697,28 @@ public static DynamoViewModel Start(StartConfiguration startConfiguration = new
             return new DynamoViewModel(startConfiguration);
         }
 
+        private void SearchDefaultNodeAutocompleteCandidates()
+        {
+            var tempSearchViewModel = new SearchViewModel(this)
+            {
+                Visible = true
+            };
+            DefaultAutocompleteCandidates = new Dictionary<string, NodeSearchElementViewModel>();
+
+            // TODO: These are basic input types in Dynamo
+            // This should be only served as a temporary default case.
+            var queries = new List<string>() { "String", "Number Slider", "Integer Slider", "Number", "Boolean", "Watch", "Watch 3D", "Python Script" };
+            foreach (var query in queries)
+            {
+                var foundNode = tempSearchViewModel.Search(query).Where(n => n.Name.Equals(query)).FirstOrDefault();
```

Comment:
Do we have to call the search API from `tempSearchViewModel`, just curious if this object intialization can be skipped?

Reply:
Well, my idea was to get the nodes for `DefaultNodeAutocomplete `at `DynamoViewModel `level and then pass them to the `NodeAutoCompleteSearchViewModel` instance (so in this way the search will be executed once per each node).

If you remember at Dynamo startup we create one `DynamoViewModel `but inside this we are creating two `WorkpaceViewModel `instances and inside each instance we are creating the `NodeAutoCompleteSearchViewModel `instance and the `SearchViewModel `instance, so the SearchViewModel instance is at `WorkpaceViewModel `level that's why I had to create the tempSearchViewModel instance (`SearchViewModel`) at `DynamoViewModel `level (we don't have that instance at that level) otherwise I won't be able to execute the Search for each node.

Let me know if it sounds clear to you or otherwise I can make a diagram showing the details.
Thanks